jekyll/_cci2/browser-testing.md

```diff
@@ -0,0 +1,239 @@
+      - image: circleci/python:jessie-node-browsers
+    steps:
+      - checkout
+      - run: 
+          name: Install Sauce
+          command: npm install saucelabs
+      - run: 
+          name: sauce testing
+          command: npm run-script sauce
+          environment:
+            SAUCE_USERNAME: # Refer to circleci.com/docs/2.0/env-vars documentation for info
+            SAUCE_ACCESS_KEY: # about setting up environment variables for auth secrets
+      - run: # Wait for app to be ready
+          command: wget --retry-connrefused --no-check-certificate -T 30 http://localhost:5000
+      - run: # Run selenium tests
+          command: nosetests
+      - run: # wait for Sauce Connect to close the tunnel
+          command: killall --wait sc  
+```
+
+## BrowserStack and Appium
+
+As in the Sauce Labs example above, you could replace the installation of Sauce Labs with installation of another cross-browser testing platform such as BrowserStack and set the USERNAME and ACCESS_KEY [environment variables]({{ site.baseurl }}/2.0/env-vars/) to those associated with your BrowserStack account.
+
+For mobile applications, it is possible to use Appium or an equivalent platform that also uses the WebDriver protocol by installing Appium in your job and using CircleCI [environment variables]({{ site.baseurl }}/2.0/env-vars/) for the USERNAME and ACCESS_KEY.
+
+## Debugging Browser Tests
+
+Integration tests can be hard to debug, especially when they're running on a remote machine.
+This section provides some examples of how to debug browser tests on CircleCI.
+
+### Using Screenshots and Artifacts
+
+CircleCI may be configured to collect [build artifacts]( {{ site.baseurl }}/2.0/artifacts/)
+and make them available from your build. For example, artifacts enable you to save screenshots as part of your build,
+and view them when the build finishes. You must explicitly collect those files with the `store_artifacts` step and specify the `path` and `destination`, see the [store_artifacts]( {{ site.baseurl }}/2.0/configuration-reference/#store_artifacts) section of the Configuration Reference for an example.
+
+Saving screenshots is straightforward: it's a built-in feature in WebKit and Selenium, and is supported by most test suites:
+
+*   [Manually, using Selenium directly](http://docs.seleniumhq.org/docs/04_webdriver_advanced.jsp#remotewebdriver)
+*   [Automatically on failure, using Cucumber](https://github.com/mattheworiordan/capybara-screenshot)
+*   [Automatically on failure, using Behat and Mink](https://gist.github.com/michalochman/3175175)
+
+### Using a Local Browser to Access HTTP server on CircleCI
+
+If you are running a test that runs an HTTP server on CircleCI, sometimes it is helpful to use a browser running on your local machine to debug a
+failing test. Setting this up is easy with an SSH-enabled
+run.
+
+First, run an SSH build using the Rerun Job with SSH button on the Builds page of the CircleCI app. The command to log into
```

Comment:
Should this and other instructions below be using the "Steps" format other docs are using?

Reply:
yes, fixed ( I think ) there might be some numbering issues with the multi-line code blocks.